Assets/HoloToolkit/Utilities/Scripts/HeadsUpDirectionIndicator.cs

```diff
@@ -0,0 +1,313 @@
+    // create your own pointer object which can by any 3D game object. You'll need to adjust
+    // the depth, margin and pivot variables to affect the right appearance. After that you
+    // simply need to specify the "targetObject" and then you should be set.
+    // 
+    // This script assumes your point object "aims" along its local up axis and orients the
+    // object according to that assumption.
+    public class HeadsUpDirectionIndicator : MonoBehaviour
+    {
+        // Use as a named indexer for Unity's frustum planes. The order follows that layed
+        // out in the API documentation. DO NOT CHANGE ORDER unless a corresponding change
+        // has been made in the Unity API.
+        private enum FrustumPlanes
+        {
+            Left = 0,
+            Right,
+            Bottom,
+            Top,
+            Near,
+            Far
+        }
+
+        [Tooltip("The object the direction indicator will point to.")]
+        public GameObject TargetObject;
+
+        [Tooltip("The camera depth at which the indicator rests.")]
+        public float Depth;
+
+        [Tooltip("The point around which the indicator pivots. Should be placed at the model's 'tip'.")]
+        public Vector3 Pivot;
+
+        [Tooltip("The object used to 'point' at the target.")]
+        public GameObject PointerPrefab;
+
+        [Tooltip("Determines what percentage of the visible field should be margin.")]
+        [Range(0.0f, 100.0f)]
+        public float IndicatorMarginPercent;
+
+        [Tooltip("Debug draw the planes used to calculate the pointer lock location.")]
+        public bool DebugDrawPointerOrientationPlanes;
+
+        private GameObject pointer;
+
+        private void Start()
+        {
+            Depth = Mathf.Clamp(Depth, Camera.main.nearClipPlane, Camera.main.farClipPlane);
+
+            pointer = GameObject.Instantiate(PointerPrefab);
+
+            // We create the effect of pivoting rotations by parenting the pointer and
+            // offsetting its position.
+            pointer.transform.parent = transform;
+            pointer.transform.position = -Pivot;
+        }
+
+        // Update the direction indicator's position and orientation every frame.
+        private void Update()
+        {
+            // No object to track?
+            if (TargetObject == null)
+            {
+                // bail out early.
+                return;
+            }
+            else
+            {
+                // The top, bottom and side frustum planes are used to restrict the movement
+                // of the pointer.
+
+                float marginFactor = IndicatorMarginPercent / 100.0f;
+
+                // Here we adjust the Camera's frustum planes to place the cursor in a smaller
+                // volume, thus creating the effect of a "margin"
+                Plane[] indicatorVolume = GeometryUtility.CalculateFrustumPlanes(Camera.main);
```

Comment:
if we can cache the results of CalculateFrustrumPlanes so it only needs to be called once per frame (rather than once per frame per direction indicator) there would be a minor perf win.